### ISSUE TITLE
plugin/kubernetes: NXDOMAIN for default.pod.cluster.local is wrong

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#1228]
It should be NODATA as it is a non-terminal in pods verified mode:

~~~
dig +search srv 10-40-4-3.default.pod.cluster.local.
;10-40-4-3.default.pod.cluster.local. IN	SRV

;; ANSWER SECTION:
10-40-4-3.default.pod.cluster.local. 0 IN SRV	0 100 0 10-40-4-3.default.pod.cluster.local.

;; ADDITIONAL SECTION:
10-40-4-3.default.pod.cluster.local. 0 IN A	10.40.4.3
~~~
and

~~~
dig +search srv default.pod.cluster.local.
 ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 59987
~~~

Probably hit this for `default.svc.cluster.local` as well, but there is a default svc running (always?)

**Comment by miekg** [comment]
Gah. Looks like this needs to be special cased in `findPods`

**Comment by johnbelamaric** [comment]
default is the namespace not the service. which I guess is why you are doing #1229

**Comment by miekg** [comment]
uh, yes. And yes :)

**Comment by miekg** [comment]
Fixed via #1229
